loader/src/sources/albertsons/index.js

```diff
@@ -260,60 +271,54 @@ async function getData(states) {
+    if (!unknown && (!adult || !pediatric)) {
+      warn(
+        "Trying to merge locations other than an adult and pediatric!",
+        group
+      );
+      return null;
+    }
 
-  return Object.values(groups)
-    .map((group) => {
-      if (group.length === 1) return group[0];
-
-      const adult = group.find((l) => l.meta.booking_url_adult);
-      const pediatric = group.find((l) => l.meta.booking_url_pediatric);
-
-      // If a location had no available vaccines and had no special naming
-      // prefix, we won't know whether it is adult or pediatric. We can't know
-      // whether this is a problem or not, so always allow it.
-      const unknown = group.find((l) => !l.availability.products);
-      if (!unknown && (!adult || !pediatric)) {
-        warn(
-          "Trying to merge locations other than an adult and pediatric!",
-          group
-        );
-        return null;
-      }
+    const result = Object.assign({}, ...group);
+    result.meta = {
+      ...adult.meta,
+      booking_url_adult: adult?.booking_url,
+      booking_url_pediatric: pediatric?.booking_url,
+    };
+    result.booking_url = GENERIC_BOOKING_URL;
+    result.external_ids = getUniqueExternalIds(
+      group.flatMap((l) => l.external_ids)
+    );
+    const products = [
+      ...new Set(group.flatMap((l) => l.availability.products).filter(Boolean)),
+    ];
+    result.availability.products = products.length ? products : undefined;
+    result.availability.available = Available.unknown;
+    if (group.some((l) => l.availability.available === Available.yes)) {
+      result.availability.available = Available.yes;
+    } else if (group.every((l) => l.availability.available === Available.no)) {
+      result.availability.available = Available.no;
+    }
 
-      const result = Object.assign({}, ...group);
-      result.meta = {
-        ...adult.meta,
-        booking_url_adult: adult?.booking_url,
-        booking_url_pediatric: pediatric?.booking_url,
-      };
-      result.booking_url = GENERIC_BOOKING_URL;
-      result.external_ids = getUniqueExternalIds(
-        group.flatMap((l) => l.external_ids)
-      );
-      const products = [
-        ...new Set(
-          group.flatMap((l) => l.availability.products).filter(Boolean)
-        ),
-      ];
-      result.availability.products = products.length ? products : undefined;
-      result.availability.available = Available.unknown;
-      if (group.some((l) => l.availability.available === Available.yes)) {
-        result.availability.available = Available.yes;
-      } else if (
-        group.every((l) => l.availability.available === Available.no)
-      ) {
-        result.availability.available = Available.no;
-      }
+    return result;
+  }
 
-      return result;
-    })
-    .filter(Boolean);
+  const groups = groupDuplicateLocations(
+    formatted,
+    ["albertsons_store_number"],
+    true
+  );
+  return groups.map((group) => mergeLocations([...group])).filter(Boolean);
```

Comment:
Now that a location can have multiple `albertsons_store_number` IDs, the old grouping logic doesn’t work reliably. I ported the more robust logic from the dedupe script in the server to use here.